drivers/modem/hl7800.c

```diff
@@ -0,0 +1,4900 @@
+
+/* The access point name (and username and password) are stored in the modem's
+ * non-volatile memory.
+ */
+int32_t mdm_hl7800_update_apn(char *access_point_name)
+{
+	int ret = -EINVAL;
+
+	hl7800_lock();
+	wakeup_hl7800();
+	ictx.last_socket_id = 0;
+	ret = write_apn(access_point_name);
+	allow_sleep(true);
+	hl7800_unlock();
+
+	if (ret >= 0) {
+		/* After a reset the APN will be re-read from the modem
+		 * and an event will be generated.
+		 */
+		k_delayed_work_submit_to_queue(&hl7800_workq,
+					       &ictx.mdm_reset_work, K_NO_WAIT);
+	}
+	return ret;
+}
+
+bool mdm_hl7800_valid_rat(uint8_t value)
+{
+	if ((value == MDM_RAT_CAT_M1) || (value == MDM_RAT_CAT_NB1)) {
+		return true;
+	}
+	return false;
+}
+
+int32_t mdm_hl7800_update_rat(enum mdm_hl7800_radio_mode value)
+{
+	int ret = -EINVAL;
+
+	if (value == ictx.mdm_rat) {
+		/* The set command will fail (in the modem)
+		 * if the RAT isn't different.
+		 */
+		return 0;
+	} else if (!mdm_hl7800_valid_rat(value)) {
+		return ret;
+	}
+
+	hl7800_lock();
+	wakeup_hl7800();
+	ictx.last_socket_id = 0;
+
+	if (value == MDM_RAT_CAT_M1) {
+		SEND_AT_CMD_ONCE_EXPECT_OK("AT+KSRAT=0");
+	} else { /* MDM_RAT_CAT_NB1 */
+		SEND_AT_CMD_ONCE_EXPECT_OK("AT+KSRAT=1");
+	}
+
+error:
+	if (ret >= 0) {
+		/* Changing the RAT causes the modem to reset. */
+		ret = modem_boot_handler("RAT changed");
+	}
+
+	allow_sleep(true);
+	hl7800_unlock();
+
+	/* A reset and reconfigure ensures the modem configuration and
+	 * state are valid
+	 */
+	if (ret >= 0) {
+		k_delayed_work_submit_to_queue(&hl7800_workq,
+					       &ictx.mdm_reset_work, K_NO_WAIT);
+	}
+
+	return ret;
+}
+
+#ifdef CONFIG_NEWLIB_LIBC
+int32_t mdm_hl7800_get_local_time(struct tm *tm, int32_t *offset)
+{
+	int ret;
+
+	ictx.local_time_valid = false;
+
+	hl7800_lock();
+	wakeup_hl7800();
+	ictx.last_socket_id = 0;
+	ret = send_at_cmd(NULL, "AT+CCLK?", MDM_CMD_SEND_TIMEOUT, 0, false);
+	allow_sleep(true);
+	if (ictx.local_time_valid) {
+		memcpy(tm, &ictx.local_time, sizeof(struct tm));
+		memcpy(offset, &ictx.local_time_offset, sizeof(*offset));
+	} else {
+		ret = -EIO;
+	}
+	hl7800_unlock();
+	return ret;
+}
+#endif
+
+void mdm_hl7800_generate_status_events(void)
+{
+	hl7800_lock();
+	generate_startup_state_event();
+	generate_network_state_event();
+	generate_sleep_state_event();
+#ifdef CONFIG_MODEM_HL7800_FW_UPDATE
+	generate_fota_state_event();
+#endif
+	event_handler(HL7800_EVENT_RSSI, &ictx.mdm_ctx.data_rssi);
+	event_handler(HL7800_EVENT_SINR, &ictx.mdm_sinr);
+	event_handler(HL7800_EVENT_APN_UPDATE, &ictx.mdm_apn);
+	event_handler(HL7800_EVENT_RAT, &ictx.mdm_rat);
+	event_handler(HL7800_EVENT_BANDS, ictx.mdm_bands_string);
+	event_handler(HL7800_EVENT_ACTIVE_BANDS, ictx.mdm_active_bands_string);
+	event_handler(HL7800_EVENT_REVISION, ictx.mdm_revision);
+	hl7800_unlock();
+}
+
+static int send_data(struct hl7800_socket *sock, struct net_pkt *pkt)
+{
+	int ret;
+	struct net_buf *frag;
+	char dstAddr[sizeof("###.###.###.###")];
```

Comment:
And here. In general, there are a few more places in this file where camelCase variables mixed in, would be good to sort them out before merge.